tests/suites/helpers.function

```diff
@@ -150,9 +150,41 @@ typedef struct data_tag
     mbedtls_exit( 1 );                                             \
 }
 
-/** Return the number of elements of a static or stack array. */
-#define ARRAY_LENGTH( array )                   \
+#if defined(__GNUC__)
+/* Test if arg and &(arg)[0] have the same type. This is true if arg is
+ * an array but not if it's a pointer. */
+#define IS_ARRAY_NOT_POINTER( arg )                                     \
+    ( ! __builtin_types_compatible_p( __typeof__( arg ),                \
+                                      __typeof__( &( arg )[0] ) ) )
+#else
+/* On platforms where we don't know how to implement this check,
+ * omit it. Oh well, a non-portable check is better than nothing. */
+#define STATIC_ASSERT_EXPR_IF_POSSIBLE( const_expr ) ( (void) 0 )
```

Comment:
Where is `IS_ARRAY_NOT_POINTER` defined for non GNUC platforms? Should `STATIC_ASSERT_EXPR_IF_POSSIBLE` instead be `IS_ARRAY_NOT_POINTER`? Where is `STATIC_ASSERT_EXPR_IF_POSSIBLE` used?

Reply:
Leftover from a slightly different design that I abandoned without committing. I have amended the corresponding commit.